tests/testthat/test-methods-epi_archive.R

```diff
@@ -0,0 +1,67 @@
+library(dplyr)
+
+ea <- archive_cases_dv_subset$clone()
+
+# epix_as_of tests
+test_that("epix_as_of behaves identically to as_of method",{
+  expect_identical(epix_as_of(ea,max_version = min(ea$DT$version)),
+                   ea$as_of(max_version = min(ea$DT$version)))
+})
+
+test_that("Errors are thrown due to bad as_of inputs",{
+  # max_version cannot be of string class rather than date class
+  expect_error(ea$as_of("2020-01-01"))
+  # max_version cannot be later than latest version
+  expect_error(ea$as_of(as.Date("2025-01-01")))
+  # max_version cannot be a vector
+  expect_error(ea$as_of(c(as.Date("2020-01-01"), as.Date("2020-01-02"))))
+})
+
+test_that("Warning against max_version being same as edf's max version",{
+  expect_warning(ea$as_of(max_version = max(ea$DT$version)))
+  expect_warning(ea$as_of(max_version = min(ea$DT$version)),NA)
+})
+
+test_that("as_of properly grabs the data",{
+  df_as_of <- ea %>%
+    epix_as_of(max_version = as.Date("2020-07-01")) %>%
+    na.omit() %>%
+    as.data.frame()
+    
+  df_filter <- ea$DT %>%
+    filter(version == as.Date("2020-07-01")) %>%
+    na.omit() %>%
+    select(-version) %>%
+    as.data.frame()
+  
+  expect_equal(df_as_of[1:4],df_filter)
+})
```

Comment:
Please replace this with a hand-crafted example that tests both that `as_of` incorporates data from `max_version` as well as uses LVCF.  E.g., have an observation that's revised from 1 to 2 to 3 to 4, and as for it `as_of` between the 3 and 4 versions; it should output 3 (and not give any extra junk related to 1 or 2, nor somehow see 4).

Reply:
I just did so.